File: code/c++/function2.c
#include <iostream>
#include <vector>
#include <cmath>
#include <cfloat>
#include <iomanip>
using namespace std;
 
vector<vector<int>> function2(int n) {   
    vector<vector<int>> t;
    vector<int> v0 = {1};
    t.push_back(v0);
    for (int i = 1; i < n; i++) {
        vector<int> vi;
        vi.push_back(1);
        for (int j = 1; j < i; j++) {
            vi.push_back(t[i - 1][j - 1] + t[i - 1][j]);
        }
        vi.push_back(1);
        t.push_back(vi);       
    }
    return t;
}

File: code/python/function2.py
def foo(r, c):
    if c in (0, r):
        return 1
    return foo(r - 1, c - 1) + foo(r - 1, c)


def function2(n):
    return [[foo(r, c) for c in range(r + 1)] for r in range(n)][n - 1]
